src/Servers/Kestrel/Core/src/Features/IRejectedRequestDetailsFeature.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Features
+{
+    /// <summary>
+    /// Provides information about rejected HTTP requests.
+    /// </summary>
+    public interface IRejectedRequestDetailsFeature
+    {
+        /// <summary>
+        /// Synchronously retrieves the exception associated with the rejected HTTP request.
+        /// </summary>
+        Microsoft.AspNetCore.Http.BadHttpRequestException? Reason { get; set; }
```

Comment:
The name `Reason` here is a carryover from when this was just a string that captured the reason. I think it should be renamed. I'm thinking `RejectedRequestException`? Other suggestions?

Reply:
Re-thinking this a bit, maybe this should be:

```C#
namespace Microsoft.AspNetCore.Http
{
    public interface IBadRequestExceptionFeature
    {
        BadHttpRequestException? Error { get; }
    }
}
```

I also think we said the feature itself shouldn't have properties that are nullable if the feature itself only exists when there's a bad request. We might be able to skip that here.

I also think it should be get only.

Reply:
> Maybe it should also move back into Microsoft.AspNetCore.Http.Features if it has a more generic name and uses the more generic exception. The only downside there is that you might want to implement it in IIS as well (but lets punt for now).

Now that we've changed the type to `BadHttpRequestException` instead of string, moving it back to `Microsoft.AspNetCore.Http.Features` is problematic... The exception type is defined in `Microsoft.AspNetCore.Http.Abstractions` which we can't reference from `Microsoft.AspNetCore.Http.Features` AFAICT.

Reply:
Hmm, maybe we should make sure this type is movable to Features even if we don't move it as a future proofing measure.

Reply:
We could also move this to Http.Abstractions. I think there was a similar issue with [IRouteValuesFeature](https://github.com/dotnet/aspnetcore/blob/6eaba08ae9e31e5404c39fcc824503b89dd53b69/src/Http/Http.Abstractions/src/Routing/IRouteValuesFeature.cs), and that's where that lives.